lib/std/heap/memory_pool.zig

```diff
@@ -0,0 +1,188 @@
+const std = @import("../std.zig");
+
+const debug_mode = @import("builtin").mode == .Debug;
+
+pub const MemoryPoolError = error{OutOfMemory};
+
+/// A memory pool that can allocate objects of a single type very quickly.
+/// Use this when you need to allocate a lot of objects of the same type,
+/// because It outperforms general purpose allocators.
+pub fn MemoryPool(comptime Item: type) type {
+    return MemoryPoolAligned(Item, @alignOf(Item));
+}
+
+/// A memory pool that can allocate objects of a single type very quickly.
+/// Instead of aligning the objects to their natural alignment, this
+/// implementation allows configuring a user-defined alignment.
+///
+/// Use this when you need to allocate a lot of objects of the same type,
+/// because It outperforms general purpose allocators.
+pub fn MemoryPoolAligned(comptime Item: type, comptime alignment: u29) type {
+    if (@alignOf(Item) == alignment) {
+        return MemoryPoolExtra(Item, .{});
+    } else {
+        return MemoryPoolExtra(Item, .{ .alignment = alignment });
+    }
+}
+
+pub const Options = struct {
+    /// The alignment of the memory pool items. Use `null` for natural alignment.
+    alignment: ?u29 = null,
+
+    /// If `true`, the memory pool can allocate additional items after a initial setup.
+    /// If `false`, the memory pool will not allocate further after a call to `initPreheated`.
+    growable: bool = true,
```

Comment:
This can already be configured by changing whether the underlying allocator is growable or not.

Reply:
Not really, as the internal arena will allocate in larger chunks and not a specific amount of items. But i guess there's some space for improvements, true.

Reply:
Oh I see, I thought the underlying allocator was just a generic Allocator. Carry on.

Reply:
It uses a generic allocator, but has an internal arena to have a simpler way of freeing the allocated memory